src/allocator/allocator.h

```diff
@@ -42,4 +46,7 @@ void Mem_Pop(i32 bytes);
 #define Temp_Reserve(p, l) Mem_Reserve(EAlloc_Temp, (p), (l))
 #define Temp_Grow(p, l) Mem_Grow(EAlloc_Temp, (p), (l))
 
+#define Script_Reserve(p, l) Mem_Reserve(EAlloc_Script, (p), (l))
+#define ScriptTemp_Grow(p, l) Mem_Grow(EAlloc_Script, (p), (l))
```

Comment:
Ah right these are the array macros I was thinking of earlier: Reserve, Grow, PopSwap.
Might want to name these new ones ScriptMem_X? ScriptTemp might be a typo?
If this will only be used by lua we could just skip adding the new macros if you want.

Reply:
Yeah ScriptTemp is a typo. And yeah let's skip adding the other macros until we need them.